lib/music/MusicInterface.js

```diff
@@ -0,0 +1,219 @@
+		 * The array of lavalink specified songs
+         * @since 2.0.1
+		 * @type {Array<Song>}
+		 */
+        this.queue = [];
+        /**
+		 * The boolean to display if the client is currently playing
+         * @since 2.0.1
+		 * @type {Boolean}
+		 */
+        this.playing = false;
+        /**
+		 * The array of lavalink specified songs
+         * @since 2.0.1
+		 * @type {Boolean}
+		 */
+        this.paused = false;
+        /**
+		 * The array of lavalink specified songs
+         * @since 2.0.1
+		 * @type {number}
+		 */
+        this.volume = guild.configs.musicVolume;
+    }
+
+    /**
+	 * Returns the guilds voice channel
+     * @since 2.0.1
+	 * @returns {Discord.voiceChannel}
+	 * @readonly
+	 */
+    get voiceChannel() {
+        return this.guild.me.voiceChannel;
+    }
+
+    /**
+	 * Returns the guilds current player
+     * @since 2.0.1
+	 * @returns {Object<Player>}
+	 * @readonly
+	 */
+    get player() {
+        return this.client.lavalink.get(this.guild.id) || null;
+    }
+
+    /**
+	 * The method to add a LavalinkTrack to the queue
+     * @since 2.0.1
+	 * @param {LavalinkTrack} songdata The LavalinkTrack data to be added and formated for the queue to handle
+	 * @param {Discord.GuildMember} requester The person who requested the addition of this song
+	 * @returns {Song}
+	 */
+    add(songdata, requester) {
+        const song = new Song(songdata, requester);
+        this.queue.push(song);
+        return song;
+    }
+
+    /**
+	 * Makes the bot join a given voice channel with the optimal region and into the specified voiceChannel
+     * @since 2.0.1
+	 * @param {Discord.voiceChannel} voiceChannel The voice channel the player should join
+	 * @returns {Object<Player>}
+	 */
+    join(voiceChannel) {
+        return this.client.lavalink.join({
+            guild: this.guild.id,
+            channel: voiceChannel.id,
+            host: this.client.lavalink.getIdealHost(this.guild.region)
+        }, { selfdeaf: this.guild.configs.selfDeaf ? this.guild.configs.selfDeaf : true });
+    }
+
+    /**
+	 * Makes the bot leave the voiceChannel it is currently in and set the playing status to false
+     * @since 2.0.1
+	 * @param {boolean} [force=true] If the bot should force leave the curreny channel
+	 * @returns {this}
+	 */
+    async leave(force = true) {
+        if (this.playing || this.player || force) this.player.stop();
+        await this.client.lavalink.leave(this.guild.id);
+        this.playing = false;
+        return this;
+    }
+
+    /**
+	 * Starts the player and checks if all criteria is met otherwise it will throw the warning but if it works it will start the player
+     * @since 2.0.1
+	 * @returns {Player}
+	 */
+    async play() {
+        const check = !this.voiceChannel ? "Well, Well, Well, It seems theres no where to play the sick tunes" : !this.player ? "Well it seems I cant find the cable for my 🎧" : this.queue.length === 0 ? "I have no records to play so please provide some" : null;
```

Comment:
no, this is wrong.
These checks should be in the play command the the way you're handling this is idiotic

Reply:
Hello?